packages/astro/src/astro-components/control/Protect.astro

```diff
@@ -1,38 +1,20 @@
 ---
-import type {
-  CheckAuthorizationWithCustomPermissions,
-  OrganizationCustomPermissionKey,
-  OrganizationCustomRoleKey,
-} from '@clerk/types';
+import ProtectCSR from './ProtectCSR.astro';
+import ProtectSSR from './ProtectSSR.astro';
 
-type Props =
-  | {
-      condition?: never;
-      role: OrganizationCustomRoleKey;
-      permission?: never;
-    }
-  | {
-      condition?: never;
-      role?: never;
-      permission: OrganizationCustomPermissionKey;
-    }
-  | {
-      condition: (has: CheckAuthorizationWithCustomPermissions) => boolean;
-      role?: never;
-      permission?: never;
-    }
-  | {
-      condition?: never;
-      role?: never;
-      permission?: never;
-    };
+import { isStaticOutput } from "virtual:@clerk/astro/config";
+import type { ProtectProps } from '../../types';
```

Comment:
Does this work because we now expose `@clerk/astro/types` ?

Reply:
We are only copying the `types.ts` file to so that it can still be resolvable. We don't export the types in `package.json` still, though. I don't see any helpful type to export 🤔